Counterexamples/TopologistsSineCurve.lean

```diff
@@ -0,0 +1,211 @@
+/-
+Copyright (c) 2025 Daniele Bolla. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniele Bolla, David Loeffler
+-/
+import Mathlib.Analysis.SpecialFunctions.Trigonometric.Inverse
+import Mathlib.Topology.Connected.PathConnected
+
+/-!
+# The "topologist's sine curve" is connected but not path-connected
+
+We give an example of a closed subset `T` of `ℝ × ℝ` which is connected but not path-connected: the
+closure of the set `{ (x, sin x⁻¹) | x ∈ Ioi 0 }`.
+
+## Main results
+
+* `TopologistsSineCurve.T_is_conn`: the set `T` is connected.
+* `TopologistsSineCurve.T_is_not_path_conn`: the set `T` is not path-connected.
+
+This formalization is part of the UniDistance Switzerland bachelor thesis of Daniele Bolla. A
+similar result has also been independently formalized by Vlad Tsyrklevich
+(https://leanprover.zulipchat.com/#narrow/channel/113489-new-members/topic/golf.20request.3A.20Topologist's.20sine.20curve).
+-/
+
+open Topology Filter Set Real
+
+namespace TopologistsSineCurve
+
+/-- The topologist's sine curve, i.e. the graph of `y = sin (x⁻¹)` for `0 < x`. -/
+def S : Set (ℝ × ℝ) := (fun x ↦ (x, sin x⁻¹)) '' Ioi 0
+
+/-- The vertical line segment `{ (0, y) | -1 ≤ y ≤ 1 }`, which is the set of limit points of `S`
+not contained in `S` itself. -/
+def Z : Set (ℝ × ℝ) := (fun y ↦ (0, y)) '' Icc (-1) 1
+
+/-- The union of `S` and `Z` (which we will show is the closure of `S`). -/
+def T : Set (ℝ × ℝ) := S ∪ Z
+
+/-- A sequence of `x`-values tending to 0 at which the sine curve has a given `y`-coordinate. -/
+noncomputable def x_seq (y : ℝ) (k : ℕ) := 1 / (arcsin y + (k + 1) * (2 * π))
```

Comment:
```suggestion
noncomputable def xSeq (y : ℝ) (k : ℕ) := 1 / (arcsin y + (k + 1) * (2 * π))
```
I know it's for counterexamples, so we shouldn't be finicky on names, but following the naming convention here doesn't hurt and it helps to make lemma names more informative.